src/core/server/plugins/plugins_system.test.ts

```diff
@@ -569,12 +569,9 @@ describe('start', () => {
     await pluginsSystem.setupPlugins(setupDeps);
     const pluginsStart = await pluginsSystem.startPlugins(startDeps);
     expect(pluginsStart).toBeInstanceOf(Map);
-    expect(opensearch.client.asInternalUser.cat.plugins).toHaveBeenCalledTimes(1);
-    const log = logger.get.mock.results[0].value as jest.Mocked<Logger>;
-    expect(log.warn).toHaveBeenCalledTimes(0);
   });
 
-  it('validates opensearch plugin installation and does not error out when plugin is not installed', async () => {
+  it('validates plugin start when opensearch plugin dependency is not installed', async () => {
```

Comment:
We should add a test for incompatibilities as well as test when a plugin chooses to not start or start.

Reply:
I'm scoping this PR to add service to expose API to verify cross compatibility of plugins with their counterparts. Below enhancements and its tests will be addressed in a follow-up PR:

A) Ability to read the plugin states (Enabled/disabled/do_not_bundle) and based on that whether to include/exclude plugins from the bootstrap.
B) Notify the OSD plugins if one of their deps becomes incompatible during runtime. This needs hooks to be identified to embed this new service when engine state changes. For now, plugins can call the cross compatibility service API explicitly during runtime to get to know the compatibility state.